core/opendaq/device/include/opendaq/device_info_config.h

```diff
@@ -187,6 +187,18 @@ DECLARE_OPENDAQ_INTERFACE(IDeviceInfoConfig, IDeviceInfo)
      * @param uuid The unique ID of a system.
      */
     virtual ErrCode INTERFACE_FUNC setSystemUuid(IString* uuid) = 0;
+
+    /*!
+     * @brief Add protocol to the list of support capabilities
+     * @param serverCapability The supported protocol.
+     */
+    virtual ErrCode INTERFACE_FUNC addServerCapability(IServerCapability* serverCapability) = 0;
+
+    /*!
+     * @brief Remove protocol from the list of support capabilities
+     * @param serverCapability The supported protocol.
+     */
+    virtual ErrCode INTERFACE_FUNC removeServerCapability(IServerCapability* serverCapability) = 0;
```

Comment:
This should be on `IDeviceInfoInternal`, as it should never be called by a device vendor, but only from server modules.